fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -878,8 +878,8 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
 
     private fun setupTheme() {
         themeManager = createThemeManager()
-        // ExternalAppBrowserActivity handles it's own theming as it can be customized.
-        if (this !is ExternalAppBrowserActivity) {
+        // ExternalAppBrowserActivity exclusively handles it's own theming unless in private mode.
+        if (this !is ExternalAppBrowserActivity || browsingModeManager.mode.isPrivate) {
```

Comment:
Hmmm, we might run into a race condition here with the updates in https://github.com/mozilla-mobile/firefox-android/pull/5502 because the mode gets set asynchronously. That's my problem to solve later though 😂